### PR TITLE
SIMSBIOHUB-72: Send observations input file to Biohub

### PR DESCRIPTION
**Pull request by anissa-agahchen**
# Overview

## Links to Jira tickets

- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-72

## Description of relevant changes

- added a function to send the observation input data and send to Biohub

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

**Review by al-rosenthal — DISMISSED** [review]
☀️

File: api/src/services/platform-service.ts [hunk 1]
```diff
@@ -633,6 +635,80 @@ export class PlatformService extends DBService {
...
+   * @memberof PlatformService
+   */
+  async submitSurveyObservationInputDataToBiohub(surveyId: number, dataPackageId: string) {
+    const surveyService = new SurveyService(this.connection);
+    const occurrenceSubmissionData = await surveyService.getLatestSurveyOccurrenceSubmission(surveyId);
+
+    if (!occurrenceSubmissionData?.output_key) {
+      throw new ApiGeneralError('Failed to submit survey to BioHub', ['Occurrence record has invalid s3 output key']);
+    }
+    const s3File = await getFileFromS3(occurrenceSubmissionData.output_key);
```
Comment (NickPhura) [inline api/src/services/platform-service.ts]: Should this not be `input_key`?

Comment (NickPhura) [inline api/src/services/platform-service.ts]: This is being called here, but `s3File` is never used. In the `_makeArtifactFromObservationInputData` function we make this call again.

Comment (anissa-agahchen) [reply api/src/services/platform-service.ts]: nice catch

File: api/src/services/platform-service.ts [hunk 2]
```diff
@@ -633,6 +635,77 @@ export class PlatformService extends DBService {
...
+   *
+   * @param {number} surveyId
+   * @param {string} dataPackageId
+   * @memberof PlatformService
+   */
+  async submitSurveyObservationInputDataToBiohub(surveyId: number, dataPackageId: string) {
+    const surveyService = new SurveyService(this.connection);
+    const occurrenceSubmissionData = await surveyService.getLatestSurveyOccurrenceSubmission(surveyId);
+
+    console.log('occurrenceSubmissionData:', occurrenceSubmissionData);
```
Comment (al-rosenthal) [inline api/src/services/platform-service.ts]: nit

**Review by al-rosenthal — DISMISSED** [review]
🐎

**Review by NickPhura — APPROVED** [review]
🐃

**Review by curtisupshall — APPROVED** [review]
Code looks good to me 😊

**Pull request merged**